### DISCUSSION TITLE
Only 'cpp' for C preprocessing is supported now

### DISCUSSION THREAD

**Discussion by JulianKemmerer** [JulianKemmerer/PipelineC#326] [Ideas]
Add support for other preprocessors, ex. `clang -E` perhaps?

Relates to https://github.com/JulianKemmerer/PipelineC/issues/56

A few places in code 'cpp' is hard coded...

**Comment by JulianKemmerer** [comment]
Need to figure out what the clang equivalent of `cpp -MM -MG ` is

**Comment by RossBencina** [comment]
Any reason not to use pcpp as the builtin fallback? https://pypi.org/project/pcpp/

I have used it on a personal project. Originally David Beazley developed it for SWIG I think, so it has some lineage.

**Comment by JulianKemmerer** [comment]
I know of no reason not to, would be open to PRs 👍 

might touch on some other issues of how to handle pipelinec requirements
I think would just use it if its installed, otherwise old error message (as opposed to it being a required extra package?)

**Comment by JulianKemmerer** [comment]
Moving to Discussions: this is a feature idea rather than something broken in a design today. Especially if C front end effected only.
